### ISSUE TITLE
[BUG] Cannot Select Specific Examples in Specification Using `prism mock` and  `Prefer` Header

### ISSUE THREAD

**Issue by loganknecht** [stoplightio/prism#1692]
## Describe the bug

I cannot select the `example` desired when using the `prism mock ...` command and the `Prefer` header.

## To Reproduce

I have been able to successfully use the `Prefer code=<NUMBER>`, however I have not been able to successfully use the `example` portion. Like so, `Prefer example=<STRING>`.

So if I have something like this

```yaml
---
openapi: "3.0.2"

servers:
  - url: http://localhost:8080
    description: Local deployment from your computer

paths:
  /api/users/register:
    post:
      summary: "Register New User"
      description: "Register a new user"
      operationId: "registerNewUser"
      tags: ["Users"]
      requestBody:
        required: true
        content:
          application/json:
            schema:
              type: "object"
              properties:
                username:
                  type: "string"
                  minLength: 2
                  maxLength: 5
                  example: "John"
                email:
                  type: "string"
                  minLength: 4
                  maxLength: 100
                  example: "john@example.com"
                password:
                  type: "string"
                  minLength: 8
                  maxLength: 50
                  example: "abc123456"
              required:
                - username
                - email
                - password
      responses:
        "201":
          description: Successful registration of a new user
        "422":
          description: Failed to register new user
          content:
            application/json:
              schema:
                type: "object"
                properties:
                  code:
                    type: "string"
                    description: "Internal error code received"
                  message:
                    type: "string"
                    description: "The error string version of the internal error"
              examples:
                usernameIsTooShort:
                  value:
                    message: "The username is too short"
                usernameIsTooLong:
                  value:
                    message: "The username is too long"
                usernameIsEmpty:
                  value:
                    message: "A username MUST be provided"
```

It will actually use the `usernameIsTooShort` instead of the `usernameIsTooLong` example. It seems as though it's just picking from the top of the list. It doesn't matter how many other examples I have or specify.

### Without the Header
![image](https://user-images.githubusercontent.com/1308524/108435379-27e6c880-71fe-11eb-934c-b6ff132fb6a3.png)

### With the Header
![image](https://user-images.githubusercontent.com/1308524/108435431-44830080-71fe-11eb-86f6-d22a89cdebcb.png)

## Expected behavior

I expect the `usernameIsTooLong` example to be returned instead.

## Additional context

N/A

**Environment (remove any that are not applicable):**
 - Prism: 4.1.2
 - OS: Mac OS 10.15.7

**Comment by pytlesk4** [comment]
The relevant code is in here:

https://github.com/stoplightio/prism/blob/master/packages/http/src/mocker/negotiator

**Comment by loganknecht** [comment]
Why was this closed?

**Comment by chohmann** [comment]
@loganknecht sorry about that, but this bug fix was released in prism 4.2.0!
